hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/HtsSystemContract.java

```diff
@@ -108,11 +110,18 @@ private static FullResult resultOfExecuting(
 
                 if (responseCode == SUCCESS) {
                     final var output = pricedResult.fullResult().result().getOutput();
+                    var updater = (ProxyWorldUpdater) frame.getWorldUpdater();
+                    final var senderId = ((ProxyEvmAccount) updater.getAccount(frame.getSenderAddress())).hederaId();
```

Comment:
We can get the `senderId` directly from `attempt.senderId()` here instead of going through the `updater` again.